nautobot_chatops/dispatchers/webex_teams.py

```diff
@@ -112,3 +117,44 @@ def user_mention(self):
         # It also doesn't seem to let you use @mentions inside an adaptive card (block).
         # So for now we just use the user name throughout.
         return f"{self.context.get('user_name')}"
+
+    def send_large_table(self, header, rows):
+        """Send a large table of data to the user/channel.
+
+        Webex Teams has a character limit per message of 7439 characters.
+        This table is outputted at a max of 120 characters per line, but this varies based on the data in Nautobot.
+        This table is dynamically rendered up to the maximum allowable charaters for each posting.
+        """
+        row_counter = 0
+        header_not_set = True
+        while row_counter < len(rows):
+            start_row = row_counter
+            table_length = 0
+            table = Texttable(max_width=120)
+            table.set_deco(Texttable.HEADER)
+            if header_not_set:
+                table.header(header)
+            # Force all columns to be shown as text. Otherwise long numbers (such as account #) get abbreviated as 123.4e10
+            table.set_cols_dtype(["t" for item in header])
+
+            while True:
+                try:
+                    table.add_rows([rows[row_counter]], header=False)
+                except IndexError:
+                    # We're at the end of the rows being output.  Break loop
+                    break
+                table_length = len(table.draw())
+                # Length of table has exceeded maximum allowable characters by Webex.  Break loop
```

Comment:
Would it be cleaner/simpler to simply split the table when reaching a `table_length` of `webex_msg_char_limit - 120`, rather than exceeding the limit and having to back up and regenerate the table with one less row?

Reply:
I didn't think of this.  Updating the code as you're right, it's much simpler.